experimental/Origami/src/Origami.jl

```diff
@@ -0,0 +1,299 @@
+# Add your new types, functions, and methods here.
+
+
+module OrigamiHelper
+
+using ..GAP
+
+function __init__()
+    mod_p = "https://ag-weitze-schmithusen.github.io/ModularGroup/PackageInfo.g"
+    ori_p = "https://ag-weitze-schmithusen.github.io/Origami/PackageInfo.g"
+    GAP.Packages.install(mod_p)
+    GAP.Packages.install(ori_p)
+    GAP.Packages.load("Origami")
+end
+
+end
+
+include("types.jl")
+include("canonical.jl")
+include("deck_group.jl")
+include("action.jl")
+include("special_origigami.jl")
+
+@doc raw"""
+    origami(h::PermGroupElem, v::PermGroupElem)
+
+blablabla
+# Examples
+```jldoctest
+julia> h = @perm (1,2)
+(1,2)
+
+julia> v = @perm (1,2)
+(1,2)
+
+julia> o = origami(h,v)
+Origami ((1,2),(1,2))
+```
+
+"""
+function origami(h::PermGroupElem, v::PermGroupElem)
+    deg = max(degree(h), degree(v))
+    return Origami(GAP.Globals.Origami(GapObj(h), GapObj(v)), h, v, deg)
+    # TODO check for transitivity? GAP already does this and this was so slow
+    # for huge origamis
+    #perm_group = permutation_group(deg, [h, v])
+    #if transitivity(perm_group) > 0
+        
+    #end
+end
+
+function origami_disconnected(h::PermGroupElem, v::PermGroupElem, d::Integer)
+    return Origami(GAP.Globals.OrigamiNC(GapObj(h), GapObj(v), d), h, v, d)
+end
+
+function Base.:(==)(a::Origami, b::Origami)
+    # TODO rewrite this? for now use Gap equality
+    return GAP.evalstr("$(string(a)) = $(string(b))")
```

Comment:
Using `GAP.evalstr` should be a very, *very* last resort. You don't need to convert objects to strings to pass them to GAP, you can do it directly:
```suggestion
    return GapObj(a) == GapObj(b)
```